### PR TITLE
[Merged by Bors] - initial stuff for 3rd party connectors

### LINKED ISSUE #1903: Add support for 3rd party connectors
Right now to start a connector you can only start a connector that's an official infinyon docker image.

Let's allow some the ability to start the cluster with more allowable prefixes for images rather than just [`infinyon/fluvio-connect-{}`](https://github.com/infinyon/fluvio/blob/82c29cf9fbb2b020bca5fc5a1aca301461fbfaa3/crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs#L211-L224)

### LINKED ISSUE #1981: Allow restricting which managed connectors can be run on cluster

### PR DESCRIPTION
**Pull request by simlay**
Closes #1903. Closes #1981.

### CODE REVIEW

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs [hunk 1]
```diff
@@ -206,20 +236,43 @@ impl ManagedConnectorDeploymentController {
...
+                if type_.starts_with(prefix.as_str()) {
+                    match ThirdPartyConnectorSpec::from_url(type_).await {
+                        Ok(spec) => {
+                            image = Some(spec.image);
+                        }
+                        Err(e) => {
+                            info!("3rd party connector spec failed to retrieve {:?}", e);
+                        }
+                    }
+                    break;
```
Comment (sehz) [inline crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:256]: need unit test parsing and prefix checking

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs [hunk 2]
```diff
@@ -206,20 +236,43 @@ impl ManagedConnectorDeploymentController {
...
+                }
```
Comment (sehz) [inline crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:257]: add debugging for if start match fails

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs [hunk 3]
```diff
@@ -206,20 +236,43 @@ impl ManagedConnectorDeploymentController {
...
+            debug!(
+                "Checking 3rd party connector {:?} in allowed_prefixes - {:?}",
+                type_, allowed_connector_prefix
+            );
+            let mut image = None;
+            for prefix in allowed_connector_prefix {
+                if type_.starts_with(prefix.as_str()) {
+                    match ThirdPartyConnectorSpec::from_url(type_).await {
+                        Ok(spec) => {
+                            image = Some(spec.image);
```
Comment (sehz) [inline crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:250]: add logging here

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs [hunk 4]
```diff
@@ -206,20 +236,43 @@ impl ManagedConnectorDeploymentController {
...
-                ImagePullPolicy::Never,
-            ),
-            Some("latest") | None => (
-                format!("infinyon/fluvio-connect-{}:latest", mc_spec.type_),
-                ImagePullPolicy::Always,
-            ),
+            Some("dev") => (image, ImagePullPolicy::Never),
+            Some("latest") | None => (format!("{}:latest", image), ImagePullPolicy::Always),
             Some(version) => (
-                format!("infinyon/fluvio-connect-{}:{}", mc_spec.type_, version),
```
Comment (sehz) [inline crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:222]: can we move this into separate module to isolate logic?

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs [hunk 5]
```diff
@@ -206,20 +236,43 @@ impl ManagedConnectorDeploymentController {
             "--".to_string(),
             format!("--fluvio-topic={}", mc_spec.topic),
         ];
+        let type_ = &mc_spec.type_;
+        let image = if type_.starts_with("https://") {
```
Comment (sehz) [inline crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:240]: need option to disable third-party connector

**Review by sehz — CHANGES_REQUESTED** [review]
minor nits

File: crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs (hunk 5 above)
Comment (simlay) [reply crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:240]: The option is that there are no `--connector-prefix`s specified.

Comment (sehz) [reply crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:240]: where is used?

Comment (simlay) [reply crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:240]: Oh, it's used at the top level cluster start CLI. If you don't specify any `--connector-prefix`, the third party connectors are never used.

Comment (sehz) [reply crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs:240]: got it. make sense

**Comment by sehz** [comment]
CI failures

**Review by sehz — APPROVED** [review]
LGTM

**Comment by simlay** [comment]
bors try

**Comment by simlay** [comment]
bors try

**Pull request closed without merging**